jetty-http/src/main/java/org/eclipse/jetty/http/HttpURI.java

```diff
@@ -782,15 +782,14 @@ private void checkSegment(String uri, int segment, int end, boolean param)
 
         // Look for segment in the ambiguous segment index.
         Boolean ambiguous = __ambiguousSegments.get(uri, segment, end - segment);
-        if (ambiguous == Boolean.TRUE)
+        if (ambiguous != null)
         {
-            // The segment is always ambiguous.
-            _violations.add(Violation.SEGMENT);
-        }
-        else if (param && ambiguous == Boolean.FALSE)
-        {
-            // The segment is ambiguous only when followed by a parameter.
-            _violations.add(Violation.PARAM);
+            // Is the segment intrinsically ambiguous
+            if (ambiguous == Boolean.TRUE)
```

Comment:
Why do we even have `.` and `..` in the `Trie` if we only ever act on `Boolean.TRUE` returns?

Reply:
We act on FALSE as it gives a non-null return.

Reply:
Specifically, the logic is:
 + Any ambiguous segment that also has a param are `Violation.PARAM` (ie. they are ambiguous but may need to the parameter to make them so)
 + Only ambiguous segments that return `Boolean.TRUE` are `Violation.SEGMENT` (ie they are ambiguous with or without a parameter).

Reply:
Boolean.TRUE is not enum value. May be better 
`if (Boolean.TRUE.equals(ambiguous))`
https://stackoverflow.com/questions/16437236/boolean-true-myboolean-vs-boolean-true-equalsmyboolean